UefiPayloadPkg/Library/FdtParserLib/FdtParserLib.c

```diff
@@ -0,0 +1,692 @@
+          DEBUG ((DEBUG_INFO, "\n         Property(%08X)  %a", Property, TempStr));
+          DEBUG ((DEBUG_INFO, "  %X", Fdt32ToCpu (*Data32)));
+
+          Serial->BaudRate = Fdt32ToCpu (*Data32);
+        } else if (AsciiStrCmp (TempStr, "reg") == 0) {
+          Data32       = (UINT32 *)(PropertyPtr->Data);
+          Attribute    = Fdt32ToCpu (*(Data32 + 0));
+          StartAddress = Fdt32ToCpu (*(Data32 + 1));
+          DEBUG ((DEBUG_INFO, "\n         Property(%08X)  %a", Property, TempStr));
+          DEBUG ((DEBUG_INFO, " StartAddress   %016lX\n", StartAddress));
+          DEBUG ((DEBUG_INFO, " Attribute      %016lX\n", Attribute));
+
+          Serial->RegisterBase = StartAddress;
+          Serial->UseMmio      = Attribute == 1 ? FALSE : TRUE;
+        }
+      }
+    } else if (AsciiStrnCmp (NodePtr->Name, "framebuffer@", AsciiStrLen ("framebuffer@")) == 0) {
+      //
+      // Create GraphicInfo HOB.
+      //
+      GraphicsInfo = BuildGuidHob (&gEfiGraphicsInfoHobGuid, sizeof (EFI_PEI_GRAPHICS_INFO_HOB));
+      ASSERT (GraphicsInfo != NULL);
+      if (GraphicsInfo == NULL) {
+        break;
+      }
+
+      ZeroMem (GraphicsInfo, sizeof (EFI_PEI_GRAPHICS_INFO_HOB));
+
+      for (Property = FdtFirstPropertyOffset (Fdt, Node); Property >= 0; Property = FdtNextPropertyOffset (Fdt, Property)) {
+        PropertyPtr = FdtGetPropertyByOffset (Fdt, Property, &TempLen);
+        TempStr     = FdtGetString (Fdt, Fdt32ToCpu (PropertyPtr->NameOffset), NULL);
+        if (AsciiStrCmp (TempStr, "reg") == 0) {
+          Data32                        = (UINT32 *)(PropertyPtr->Data);
+          FrameBufferBase               = Fdt32ToCpu (*(Data32 + 0));
+          FrameBufferSize               = Fdt32ToCpu (*(Data32 + 1));
+          GraphicsInfo->FrameBufferBase = FrameBufferBase;
+          GraphicsInfo->FrameBufferSize = (UINT32)FrameBufferSize;
+        } else if (AsciiStrCmp (TempStr, "stride") == 0) {
+          Data32                                       = (UINT32 *)(PropertyPtr->Data);
+          GraphicsInfo->GraphicsMode.PixelsPerScanLine = Fdt32ToCpu (*Data32);
+        } else if (AsciiStrCmp (TempStr, "width") == 0) {
+          Data32                                          = (UINT32 *)(PropertyPtr->Data);
+          GraphicsInfo->GraphicsMode.HorizontalResolution = Fdt32ToCpu (*Data32);
+        } else if (AsciiStrCmp (TempStr, "height") == 0) {
+          Data32                                        = (UINT32 *)(PropertyPtr->Data);
+          GraphicsInfo->GraphicsMode.VerticalResolution = Fdt32ToCpu (*Data32);
+        } else if (AsciiStrCmp (TempStr, "format") == 0) {
+          TempStr = (CHAR8 *)(PropertyPtr->Data);
+          if (AsciiStrCmp (TempStr, "a8r8g8b8") == 0) {
+            GraphicsInfo->GraphicsMode.PixelFormat = PixelRedGreenBlueReserved8BitPerColor;
+          } else if (AsciiStrCmp (TempStr, "a8b8g8r8") == 0) {
+            GraphicsInfo->GraphicsMode.PixelFormat = PixelFormatMax;

```

Comment:
Support flow should be like below:

if (a8r8g8b8) 
  GraphicsInfo->GraphicsMode.PixelFormat = PixelRedGreenBlueReserved8BitPerColor
else if (a8b8g8r8) 
  GraphicsInfo->GraphicsMode.PixelFormat = PixelBlueGreenRedReserved8BitPerColor
else
  GraphicsInfo->GraphicsMode.PixelFormat = PixelFormatMax